src/components/Thanks/Badges/DetailSection.vue

```diff
@@ -164,7 +169,75 @@ export default {
 		},
 		hideBadgeName(badgeId) {
 			return badgeId !== this.currentBadge?.id;
-		}
+		},
+		async createUserPreferences() {
+			const createUserPreferencesMutation = gql`
+				mutation createUserPreferences($preferences: String) {
+					my {
+						createUserPreferences(userPreferences: {preferences: $preferences}) {
+							id
+							preferences
+						}
+					}
+				}
+			`;
+
+			const createUserPreferences = this.apollo.mutate({
+				mutation: createUserPreferencesMutation,
+				variables: {
+					preferences: '',
+				},
+			});
+			const response = await createUserPreferences;
+
+			return response?.data?.my?.createUserPreferences?.id;
+		},
+		async setAsGoal() {
+			try {
+				let updateUserPreferencesId = this.userPreferences?.id ?? null;
+
+				if (!updateUserPreferencesId) {
+					updateUserPreferencesId = await this.createUserPreferences();
+				}
+				const currentPreferences = this.userPreferences?.preferences ?? {};
+				const preferences = JSON.stringify({ ...currentPreferences, goal: this.currentBadgeName });
+
+				const updateUserPreferencesMutation = gql`
+					mutation UpdateUserPreferences(
+						$updateUserPreferencesId: Int!,
+						$preferences: String
+					) {
+						my {
+							updateUserPreferences(id: $updateUserPreferencesId, userPreferences: {
+								preferences: $preferences
+						}) {
+								id
+								preferences
+							}
+						}
+				}`;
+
+				const updateUserPreferences = this.apollo.mutate({
+					mutation: updateUserPreferencesMutation,
+					variables: {
+						updateUserPreferencesId,
+						preferences,
+					},
+				});
+
+				const response = await updateUserPreferences;
+				if (response.errors) {
+					throw new Error(
+						response.errors[0].extensions.code
+						|| response.errors[0].message
+					);
+				} else {
+					this.$route.push({ path: '/portfolio', query: { goal_saved: true } });
+				}
+			} catch (error) {
+				throw new Error(error);
```

Comment:
What happens if an error is thrown here? Is a tooltip shown to the user?

Reply:
I see there's now a tooltip, but what happens here when the `error` is thrown a second time? How does the app handle it?

Reply:
In that case, this throw should probably be a console.error then.